blog/02-09-2022-creating-a-provider-for-the-go-sdk.mdx

```diff
@@ -0,0 +1,383 @@
+---
+ slug: creating-a-provider-for-the-go-sdk
+ title: Creating a Provider for the golang-sdk
+ authors: [jamesmilligan]
+ tags: [golang, blog, tutorial, golang-sdk, providers]
+ unlisted: false
+---
+
+# Creating a Provider For The golang-sdk
+
+## Providers
+
+A [Provider](https://docs.openfeature.dev/docs/specification/sections/providers) is responsible for performing flag evaluation, they can be as simple as a local JSON evaluator, or act as an abstraction layer for a more complex evaluation system. Only one `Provider` can be registered at a time, and OpenFeature will no-op if one has not been defined. Before writing your own `Provider`, it is strongly recommended to familiarize yourself with the [OpenFeature spec](https://docs.openfeature.dev/docs/specification/).  
+In this tutorial I will demonstrate the steps taken to create a new `Provider` whilst conforming to the OpenFeature spec using a simple flag implementation. The flag evaluation will be handled by a simple JSON evaluator and flag configurations will be stored as environment variables.  
+
+
+The following section describes how the flag evaluator portion of this project has been constructed, to skip to the `Provider` specific implementations, [click here](#creating-a-compliant-provider).  
+
+## Creating the flag evaluator
+
+The core of any flag `Provider` is the evaluation logic used to provide the flag values from the provided metadata (referred to as the [Evaluation Context](https://docs.openfeature.dev/docs/specification/sections/evaluation-context)). For this example I have put together a very simple json evaluator. 
```

Comment:
It may be worth noting that it's possible to implement a provider that doesn't do dynamic evaluation, though it's certainly a lot less powerful.

Reply:
ive updated this paragraph, let me know what you think